### DISCUSSION TITLE
Dynamically changing datasource (SQLite)

### DISCUSSION THREAD

**Discussion by fmaclen** [prisma/prisma#15230] [Q&A] [answered]
I have an Electron app where a SQLite database can be switched by the user.
Every time the user changes the database the app updates the `DATABASE_URL` environment variable with the new path.

The issue I'm running into is that for Prisma to recognize this change I need to kill the process of the app that instantiated the `PrismaClient` and start it again.

I tried running `prisma.$disconnect()` when the DB path changes but that doesn't seem to have any effect as the client is still returning data from the original DB.

Is there another way I can _"reset"_ the PrismaClient to connect to the new DB without having to restart the entire app?

**Comment by pantharshit00** [comment]
You can try instantiating a new instance by passing the datasource url like so:
```ts
  const prisma = new PrismaClient({
    datasources:{
      db:{
        url:"<your url>"
      }
    }
  })
```

`datasource.db` might be called something else depending on how you have named the datasource in your schema.

**Reply by fmaclen** [reply]
The issue I'm describing is about managing already instantiated PrismaClient's, creating a new one doesn't address the problem.

**Reply by pantharshit00** [reply]
I don't think it is possible to change the datasource url after instantiation right now. Maybe create a feature request for this: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

**Reply by zomars** [reply]
Also creating new instances can get out of hand real fast. How do you handle this in prisma cloud?

**Comment by fmaclen** [comment, marked as answer]
Just created an issue with the feature request: https://github.com/prisma/prisma/issues/15746

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
